### DISCUSSION TITLE
Components referenced after running compiled code

### DISCUSSION THREAD

**Discussion by guilherme-marin** [mdx-js/mdx#2307] [Q&A] [answered]
### What I'm trying to achieve
I'm trying to [run](https://mdxjs.com/packages/mdx/#runfunctionbody-options) a MDX string which has a component in it. This component contains a reference to a next/link. I'm  trying to follow using [on-demand](https://mdxjs.com/guides/mdx-on-demand/) mdxjs instructions.

Here is the component:
```javascript
import Link from 'next/link';
import React from 'react'

const Component1 = () => {    
  return (
    <div>
        <Link href={"/"}>
          I&apos;m a LINK
        </Link>        
    </div>
  )
}

export default Component1
```
Here is the Page that tries to compile and run:
```javascript
import { useState, useEffect, Fragment } from "react";
//import * as runtime from 'react/jsx-runtime' // Production.
import * as runtime from "react/jsx-dev-runtime"; // Development.
import { compile, run } from "@mdx-js/mdx";
import Component1 from "../components/SimpleComponent";
import type { MDXModule } from "mdx/types";

const stringMDX = `
<Component1 />
`;

export default function Page({ code }: any) {
  const [mdxModule, setMdxModule] = useState<MDXModule>();
  const Content = mdxModule ? mdxModule?.default : Fragment;
  useEffect(() => {
    (async () => {
      setMdxModule(await run(code, runtime));
    })();
  }, [code]);


  return (
    <>
      {code && <Content components={{ Component1 }} />}
    </>
  );
}
export async function getStaticProps() {
  const code = String(
    await compile(stringMDX, {
      outputFormat: "function-body",
      development: process.env.NODE_ENV === "production" ? false : true,
    })
  );
  return { props: { code } };
}
```

### What happened
Client side error after build, however, while in dev, the error does not appear, according to the attached codesandbox.

### What I was expecting 
Navigation to work on [production](https://nex-link-clientside.vercel.app/) as it does in [development mode](https://codesandbox.io/p/github/guilherme-marin/nex-link-clientside/main?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522cli051qli000a356kuqxr0bhb%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522cli051qli000c356krkd0do9o%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cli051qli000a356kuqxr0bhb%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cli051qli0009356kndkrccxt%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522cli051qli000a356kuqxr0bhb%2522%252C%2522activeTabId%2522%253A%2522cli051qli0009356kndkrccxt%2522%257D%252C%2522cli051qli000c356krkd0do9o%2522%253A%257B%2522id%2522%253A%2522cli051qli000c356krkd0do9o%2522%252C%2522activeTabId%2522%253A%2522cli052kx900bh356kaxr7rbky%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522cli052jxh005x356k03kzam4o%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522cli052kx900bh356kaxr7rbky%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D). Am I missing something?

(PROD) Deployment Link: [https://nex-link-clientside.vercel.app/](https://nex-link-clientside.vercel.app/)
(DEV) Codesandbox Link: [next-link-clientside](https://codesandbox.io/p/github/guilherme-marin/nex-link-clientside/main?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522cli051qli000a356kuqxr0bhb%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522cli051qli000c356krkd0do9o%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cli051qli000a356kuqxr0bhb%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cli051qli0009356kndkrccxt%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522cli051qli000a356kuqxr0bhb%2522%252C%2522activeTabId%2522%253A%2522cli051qli0009356kndkrccxt%2522%257D%252C%2522cli051qli000c356krkd0do9o%2522%253A%257B%2522id%2522%253A%2522cli051qli000c356krkd0do9o%2522%252C%2522activeTabId%2522%253A%2522cli052kx900bh356kaxr7rbky%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522cli052jxh005x356k03kzam4o%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522cli052kx900bh356kaxr7rbky%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by wooorm** [comment]
I dunno who owns `mdx-poc` or where the code lives? What is it?

**Reply by guilherme-marin** [reply]
I'm sorry, I forgot to add the repo: [https://github.com/sync-adm/mdx-poc](https://github.com/sync-adm/mdx-poc).

**Reply by wooorm** [reply]
Ok, you have a repo that includes a ton of code. Unrelated dependencies? A month old fork? Now what? Spend time framing your question. Remove everything that isn’t needed. https://mdxjs.com/community/support/

**Reply by guilherme-marin** [reply]
@wooorm, I tried to add more context to my question, thanks for your attention to this thread so far.

**Comment by wooorm** [comment, marked as answer]
You get an error due to a difference between development and production.
This is explained in our guide: https://mdxjs.com/guides/mdx-on-demand/#nextjs-example.
To solve the error you see, follow that guide, and explicitly set the `development` option passed to `compile`.
